src/compiler/optimize/optimize-module.ts

```diff
@@ -59,7 +59,15 @@ export const optimizeModule = async (
   if (opts.sourceTarget === 'es5' || opts.minify) {
     minifyOpts = getTerserOptions(config, opts.sourceTarget, isDebug);
     if (config.sourceMap) {
-      minifyOpts.sourceMap = { content: opts.sourceMap };
+      minifyOpts.sourceMap = {
+        content:
+          opts.sourceMap != null
```

Comment:
Do we need to coerce this using `!=` ?

When `opts.sourceMap` is `undefined`, we end up just setting it to `undefined` again. Could this work as
```suggestion
          opts.sourceMap !== null
```

Reply:
I did the implicit comparison because if it is `undefined`, we want that to be falsy in the check so we don't try to spread `undefined` into the object... which is what was causing problems with the analysis tests before we reverted this upgrade

Reply:
Ahh makes sense. Might be worth throwing a comment in there explaining it in that case (since it got through all three of us once and me twice 😆)